torchvision/prototype/datasets/_builtin/sun397.py

```diff
@@ -0,0 +1,82 @@
+import pathlib
+from typing import Any, Dict, List, Tuple
+
+from torchdata.datapipes.iter import IterDataPipe, Mapper, Filter, LineReader, IterKeyZipper
+from torchvision.prototype.datasets.utils import Dataset, DatasetConfig, DatasetInfo, HttpResource, OnlineResource
+from torchvision.prototype.datasets.utils._internal import (
+    path_comparator,
+    hint_sharding,
+    hint_shuffling,
+    INFINITE_BUFFER_SIZE,
+    getitem,
+)
+from torchvision.prototype.features import EncodedImage, Label
+
+
+class SUN397(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "sun397",
+            homepage="https://vision.princeton.edu/projects/2010/SUN/",
+            valid_options=dict(
+                split=("train", "test"),
+                fold=tuple(str(fold) for fold in range(1, 11)),
```

Comment:
@pmeier 

In the original implementation of this dataset we decided not to include the fold parameter and also removed the split param https://github.com/pytorch/vision/pull/5164#issuecomment-1005813311. Should we keep things consistent here?

Reply:
That would ignore a lot of information the dataset provides. IIRC, the only reason we were ok to remove it, was that this was one of the last datasets in the old API and FLAVA didn't need the functionality. Given that this no longer applies, IMO we should handle this now.

Reply:
From what I recall, we decided not to include a `fold` parameter because:

- it's not clear what it would be used for for most users - FLAVA is our only known use-case ATM, and they don't need it
- we can add it in a BC way if users show a clear need for it
- The use of the `fold` parameter would set a precedent, and might conflict with other datasets like DTD which also partition the dataset, but in a different way. So by not supporting this we're also minimizing the chances of inconsitencies in our API.

I think all of these still apply today, so unless we have a strong and obvious use-case for including it, I would prefer to go with the status quo here, which would give us the maximal flexibility for future us.

Reply:
Hey @NicolasHug @pmeier , I changes the valid_options to the following:
It only have `split` options which value can be:

- "all" (default) -> this will give same datasets as old API, and since this is the default it will BC with the old API
- "train-{fold}", sample: "train-5" -> this will give dataset of train at fold=5 according to the paper
- "test-{fold}", sampple: "test-5" -> this will give dataset of test at fold=5 according to the paper

What are your thought on this?